tests/Geocoder/Tests/Provider/GoogleMapsTest.php

```diff
@@ -427,4 +427,64 @@ public function testGeocodeWithInvalidClientIdAndKeyNoSsl()
         $provider = GoogleMaps::business($this->getAdapter(), 'foo', 'bogus', null, null, false);
         $provider->geocode('Columbia University');
     }
+
+    /**
+     * @expectedException \Geocoder\Exception\NoResult
+     * @expectedExceptionMessage Could not execute query
+     */
+    public function testGeocodeWithComponentFiltersInvalidCountry()
+    {
+        $provider = new GoogleMaps($this->getAdapter(), 'fr-FR', 'Île-de-France');
+        $arrFilters = ['country' => 'ru', 'postal_code' => '75020'];
+        $provider->setComponentFilters($arrFilters);
+
+        $provider->geocode('10 avenue Gambetta, Paris, France');
+    }
+
+    /**
+     * @expectedException \Geocoder\Exception\NoResult
+     * @expectedExceptionMessage Could not execute query
+     */
+    public function testGeocodeWithComponentFiltersInvalidZipCode()
+    {
+        $provider = new GoogleMaps($this->getAdapter(), 'fr-FR', 'Île-de-France');
+        $arrFilters = ['country' => 'fr', 'postal_code' => '00000'];
+        $provider->setComponentFilters($arrFilters);
+
+        $provider->geocode('10 avenue Gambetta, Paris, France');
+    }
+
+    public function testGeocodeWithValidComponentFilters()
```

Comment:
ComponentFilters are 100% optional - so without it this test will not  be different, but it will fail 100%

Not sure how i can redo this - any suggestions are welcome.